SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp

```diff
@@ -42,11 +43,19 @@ FORCEINLINE UClass* ResolveClass(FString& ClassPath)
 
 void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
 {
-	checkf(IsSupportedClass(Class), TEXT("Could not find class in schema database: %s"), *Class->GetPathName());
 
 	TSharedRef<FClassInfo> Info = ClassInfoMap.Add(Class, MakeShared<FClassInfo>());
 	Info->Class = Class;
 
+	if (!IsSupportedClass(Class))
+	{
+		UE_LOG(LogSpatialClassInfoManager, Error, TEXT("Could not find class %s in schema database. Double-check whether replication is enabled for this class, the class is explicitly referenced from the starting scene and schema has been generated."), *Class->GetPathName());
+		UE_LOG(LogSpatialClassInfoManager, Error, TEXT("Disconnecting due to no generated schema for %s."), *Class->GetPathName());
+
+		(Cast<USpatialGameInstance>(NetDriver->GetWorld()->GetGameInstance()))->QuitGame();
```

Comment:
On further look, I don't think this warrants a dependency to `SpatialGameInstance`.

You can remove `USpatialGameInstance::QuitGame()`, and do all of the logic inline here.

Reply:
Ok, will move it if you think that `QuitGame()` will not be reusable.